subsys/usb/device_next/class/usbd_cdc_ncm.c

```diff
@@ -0,0 +1,1282 @@
+			iface, alternate, first_iface);
+
+	if (iface == first_iface + 1) {
+		LOG_DBG("set alt: %d", alternate);
+		data->itf_data_alt = alternate;
+	}
+
+	if (iface != first_iface + 1  ||  alternate == 0) {
+		LOG_DBG("Skip iface %u alternate %u", iface, alternate);
+
+		/*
+		 * reset internal status
+		 */
+		data->tx_sequence = 0;
+		data->if_state = IF_STATE_FIRST_SKIPPED;
+		return;
+	}
+
+	if (data->if_state == IF_STATE_INIT) {
+		data->if_state = IF_STATE_FIRST_SKIPPED;
+		LOG_DBG("Skip first iface enable");
+		return;
+	}
+
+	LOG_INF("enable net_if");
+	net_if_carrier_on(data->iface);
+
+	if (cdc_ncm_out_start(c_data)) {
+		LOG_ERR("Failed to start OUT transfer");
+	}
+
+	_usbd_cdc_ncm_notification_next_step(c_data);
+}   /* usbd_cdc_ncm_update */
+
+
+
+static void usbd_cdc_ncm_enable(struct usbd_class_data *const c_data)
+/**
+ * Class associated configuration is selected
+ */
+{
+	const struct device *dev = usbd_class_get_private(c_data);
+	struct cdc_ncm_eth_data *data = dev->data;
+
+	atomic_set_bit(&data->state, CDC_NCM_CLASS_ENABLED);
+	LOG_DBG("Configuration enabled");
+}   /* usbd_cdc_ncm_enable */
+
+
+
+static void usbd_cdc_ncm_disable(struct usbd_class_data *const c_data)
+/**
+ * Class associated configuration is disabled
+ */
+{
+	const struct device *dev = usbd_class_get_private(c_data);
+	struct cdc_ncm_eth_data *data = dev->data;
+
+	if (atomic_test_and_clear_bit(&data->state, CDC_NCM_CLASS_ENABLED)) {
+		net_if_carrier_off(data->iface);
+	}
+
+	atomic_clear_bit(&data->state, CDC_NCM_CLASS_SUSPENDED);
+	LOG_DBG("Configuration disabled");
+}   /* usbd_cdc_ncm_disable */
+
+
+
+static void usbd_cdc_ncm_suspended(struct usbd_class_data *const c_data)
+/**
+ * USB power management handler suspended
+ */
+{
+	const struct device *dev = usbd_class_get_private(c_data);
+	struct cdc_ncm_eth_data *data = dev->data;
+
+	LOG_DBG("");
+	atomic_set_bit(&data->state, CDC_NCM_CLASS_SUSPENDED);
+}   /* usbd_cdc_ncm_suspended */
+
+
+
+static void usbd_cdc_ncm_resumed(struct usbd_class_data *const c_data)
+/**
+ * USB power management handler resumed
+ */
+{
+	const struct device *dev = usbd_class_get_private(c_data);
+	struct cdc_ncm_eth_data *data = dev->data;
+
+	LOG_DBG("");
+	atomic_clear_bit(&data->state, CDC_NCM_CLASS_SUSPENDED);
+}   /* usbd_cdc_ncm_resumed */
+
+
+
+static int usbd_cdc_ncm_control_to_device(struct usbd_class_data *const c_data,
+		const struct usb_setup_packet *const setup,
+		const struct net_buf *const buf)
+/**
+ * USB control request handler to device
+ */
+{
+	if (setup->bRequest == SET_ETHERNET_PACKET_FILTER) {
+		LOG_INF("bRequest 0x%02x (SetPacketFilter) not implemented", setup->bRequest);
+		return 0;
+	}
+
+	if (setup->bRequest == NCM_SET_NTB_INPUT_SIZE) {
+		LOG_INF("bRequest 0x%02x (SetNtbInputSize) not implemented", setup->bRequest);
+		return 0;
+	}
+
+	LOG_WRN("usbd_cdc_ncm_control_to_device - bmRequestType 0x%02x bRequest 0x%02x unsupported",
```

Comment:
I think the logging subsystem for Zephyr currently handles printing the function name. It is a Kconfig.
https://docs.zephyrproject.org/latest/services/logging/index.html#global-kconfig-options
So probably no need to add it there.

Reply:
Yes, one can switch this on via CONFIG_LOG_FUNC_NAME_PREFIX_*.  Only switched on for DBG by default.

You will not see this message except if the host driver does something not expected.